lms/djangoapps/mobile_api/course_info/views.py

```diff
@@ -321,10 +321,17 @@ def list(self, request, **kwargs):  # pylint: disable=W0221
             request - Django request object
         """
 
-        response = super().list(request, kwargs)
+        api_version = self.kwargs.get('api_version')
+        if api_version == 'v4':
+            # Introduced a new version of the Course Blocks API to avoid breaking existing clients.
+            # The previous implementation unintentionally passed **kwargs as the positional argument to
+            # `hide_access_denial`, leading to potential issues. This new version removes that risk
+            # while preserving the original behavior for older clients.
+            response = super().list(request)
+        else:
+            response = super().list(request, kwargs)
```

Comment:
Thanks for the comment. I believe that the proposed fixed is being overly cautious. 

Regardless of API version, the only possible method that this instance of `super().list` could refer to is `BlocksInCourseView.list`. It would never make sense to pass `kwargs` (without `**`) into that method-- that's definitely a bug, and has been a bug since it was written.

The simplest fix would be to always pass in `**kwargs`:

```suggestion
    response = super().list(request, **kwargs)
```

and, in `BlocksInCourseView`, accept and ignore the extra kwargs:

```python
    def list(self, request, hide_access_denials=False, **_kwargs):
```

This way, if `hide_access_denials` is provided, it will percolate up to this method; if it is not provided, then `hide_access_denials` will remain False as expected. This behavior should be reasonable across all API versions.

Reply:
@kdmccormick I agree with you on this and my initial commit tried to adopt this logic. But this change will break existing clients consuming this api, they are not expecting gated blocks in response. After discussion with mobile team we adopted this safe approach, which will not effect current consumer and will also fix it for future versions.

Reply:
@jawad-khan  ahh okay, it makes sense that you don't want to break older mobile app versions which count on this bug.

Can you rephrase the conditional so that future API versions (v5+) do not reintroduce the buggy use of kwargs?

For example, if API versions v1-v3 depend on the buggy use of kwargs, you could write:
```python
        if api_version is None or api_version in ["v1", "v2", "v3"]:
            response = super().list(request, kwargs)
        else:  # v4+
            response = super().list(request)
```

Reply:
Done.